frontend/packages/kubevirt-plugin/src/k8s/objects/vm/vm/vm-clone.ts

```diff
@@ -0,0 +1,147 @@
+    const { metadata } = this.data;
+    const osId = getOperatingSystem(this.data);
+    const osName = getOperatingSystemName(this.data);
+
+    this.data.spec.running = startVM;
+    metadata.name = name;
+    metadata.namespace = namespace;
+    metadata.annotations = {};
+
+    if (description) {
+      metadata.annotations.description = description;
+    }
+    if (osId && osName) {
+      metadata.annotations[`${TEMPLATE_OS_NAME_ANNOTATION}/${osId}`] = osName;
+    }
+
+    this.getTemplateLabels()[TEMPLATE_VM_NAME_LABEL] = name;
+    return this;
+  }
+
+  withClonedPVCs = (persistentVolumeClaimsToClone: K8sResourceKind[]) => {
+    const pvcLookup = createBasicLookup(persistentVolumeClaimsToClone, getBasicID);
+    const name = getName(this.data);
+
+    this.getVolumes()
+      .filter(getVolumePersistentVolumeClaimName)
+      .forEach((volume) => {
+        const pvcName = getVolumePersistentVolumeClaimName(volume);
+        delete volume.persistentVolumeClaim;
+
+        const pvc = pvcLookup[`${this.oldVMNamespace}-${pvcName}`];
+
+        if (pvc) {
+          const clonedDVTemplate = new DatavolumeTemplate({
+            name: joinIDs(name, pvcName, 'clone'),
+            pvcSourceName: pvcName,
+            pvcSourceNamespace: this.oldVMNamespace,
+            accessModes: getPvcAccessModes(pvc),
+            size: getPvcStorageSize(pvc),
+            storageClassName: getPvcStorageClassName(pvc),
+          }).build();
+
+          this.getDataVolumeTemplates().push(clonedDVTemplate);
+
+          volume.dataVolume = {
+            name: getName(clonedDVTemplate),
+          };
+        }
+      });
+    return this;
+  };
+
+  withClonedDataVolumes = (dataVolumes: K8sResourceKind[]) => {
+    const dvLookup = createBasicLookup(dataVolumes, getBasicID);
+    const name = getName(this.data);
+
+    this.getVolumes()
+      .filter(getVolumeDataVolumeName)
+      .forEach((volume) => {
+        const dvName = getVolumeDataVolumeName(volume);
+        const dataVolume = dvLookup[`${this.oldVMNamespace}-${dvName}`];
+
+        if (dataVolume) {
+          const clonedDVTemplate = new DatavolumeTemplate({
+            name: joinIDs(name, dvName, 'clone'),
+            pvcSourceName: dvName,
+            pvcSourceNamespace: this.oldVMNamespace,
+            accessModes: getDataVolumeAccessModes(dataVolume),
+            size: getDataVolumeStorageSize(dataVolume),
+            storageClassName: getDataVolumeStorageClassName(dataVolume),
+          }).build();
+
+          this.getDataVolumeTemplates().push(clonedDVTemplate);
+
+          volume.dataVolume = {
+            name: getName(clonedDVTemplate),
+          };
+        }
+      });
+    return this;
+  };
+
+  build() {
+    const result = super.buildClean();
+    result.spec.template.spec.volumes = result.spec.template.spec.volumes.filter(
+      (v) => !getVolumePersistentVolumeClaimName(v),
+    ); // in case withClonedPVCs was not called
```

Comment:
Which statement does this comment refer to?

Reply:
line 142: the filtering

Reply:
moved the comment above the statement